firedrake/mg/kernels.py

```diff
@@ -49,7 +50,7 @@ def to_reference_coordinates(ufl_coordinate_element, parameters=None):
     code = {
         "geometric_dimension": cell.geometric_dimension(),
         "topological_dimension": cell.topological_dimension(),
-        "to_reference_coords_newton_step": to_reference_coords_newton_step_body(ufl_coordinate_element, parameters),
+        "to_reference_coords_newton_step": to_reference_coords_newton_step_body(ufl_coordinate_element, parameters, x0_dtype=ScalarType, dX_dtype="double"),
```

Comment:
this is OK - leave a comment explaining why it's needed.

Reply:
Looks like this didn't happen?

Reply:
We decided to get the merge in in the time we had. I should go back and add this.